native/cocos/bindings/jswrapper/MappingUtils.h

```diff
@@ -36,34 +37,151 @@ class Object;
     static bool init();
     static void destroy();
-
     static bool isValid();
 
-    static Map::iterator find(void *nativeObj);
+    /**
+     * @deprecated Use `contains` or `filter` to query or manipulate the elements of the map,
+     */
+    CC_DEPRECATED(3.7)
+    static Map::iterator find(void *v);
+    /**
+     * @deprecated Use `contains` or `filter` to query or manipulate the elements of the map,
+     */
+    CC_DEPRECATED(3.7)
+    static Map::iterator begin();
+    /**
+     * @deprecated Use `contains` or `filter` to query or manipulate the elements of the map,
+     */
+    CC_DEPRECATED(3.7)
+    static Map::iterator end();
+
     static Map::iterator erase(Map::iterator iter);
     static void erase(void *nativeObj);
+    static void erase(void *nativeObj, se::Object *);
     static void clear();
     static size_t size();
 
     static const Map &instance();
 
-    static Map::iterator begin();
-    static Map::iterator end();
+    /**
+     * @brief Return the first element of the specified key
+     */
+    template <typename T>
+    static se::Object *findFirst(T *nativeObj) {
+        auto itr = __nativePtrToObjectMap->find(nativeObj);
+        return itr == __nativePtrToObjectMap->end() ? nullptr : itr->second;
+    }
 
+    /**
+     * @brief Check if the key exists in the map
+     */
     template <typename T>
-    static std::enable_if_t<!std::is_void_v<T>, Map::iterator> find(T *v) {
-        if constexpr (std::is_const_v<T>) {
-            return find(reinterpret_cast<void *>(const_cast<std::remove_const_t<T> *>(v)));
+    static bool contains(T *nativeObj) {
+        if constexpr (std::is_void_v<T>) {
+            return __nativePtrToObjectMap->count(nativeObj) > 0;
         } else {
-            return find(reinterpret_cast<void *>(v));
+            auto *kls = JSBClassType::findClass(nativeObj);
+            auto range = __nativePtrToObjectMap->equal_range(nativeObj);
+            for (auto itr = range.first; itr != range.second; itr++) {
+                if (itr->second->_getClass() == kls) {
+                    return true;
+                }
+            }
+            return false;
         }
     }
 
+    /**
+     * @brief Iterate se::Object with specified key
+     */
+    template <typename T, typename Fn>
+    static void forEach(T *nativeObj, const Fn &func) {
+        se::Class *kls = nullptr;
+        if constexpr (!std::is_void_v<T>) {
+            kls = JSBClassType::findClass(nativeObj);
+        }
+        auto range = __nativePtrToObjectMap->equal_range(nativeObj);
+        for (auto itr = range.first; itr != range.second; itr++) {
+            if (kls != nullptr && kls != itr->second->_getClass()) {
+                continue;
+            }
+            func(itr->second);
+        }
+    }
+    /**
+     * @brief Filter se::Object* with key and se::Class value
+     */
+    template <typename T>
+    static OptionalCallback filter(T *nativeObj, se::Class *kls) {
+        se::Object *target;
```

Comment:
Why not to initialize the `target` to `nullptr` here?

Reply:
It will be set in either callbacks of findWithCallback

Reply:
fixed